src/components/PlantingSites/index.tsx

```diff
@@ -16,27 +19,34 @@ export type PlantingSitesProps = {
   reloadTracking: () => void;
 };
 
-export default function PlantingSites(props: PlantingSitesProps): JSX.Element {
-  const [initializingData] = useState<boolean>(false);
-
-  /**
-   * TODO: intialize data
-   */
-
-  // show spinner while initializing data
-  if (initializingData) {
-    return <CircularProgress sx={{ margin: 'auto' }} />;
-  }
-
-  return <PlantingSitesWrapper {...props} />;
+export default function PlantingSites({ reloadTracking }: PlantingSitesProps): JSX.Element {
+  return (
+    <Switch>
```

Comment:
Child Switches appear to route on the full path? That is, does the app ultimately see a flattened list of Routes, or does the nesting affect how things are routed?

Reply:
If the `PlantingSitesWrapper`, for instance, is not rendered, can the app see its routes?

Reply:
That is correct, the routing is on the full path.
The root App.tsx has a route `/planting-sites` (non-exact match, just needs to start with that) - which renders the PlantingSitesWrapper. Once that is rendered, the finer route handling happens.
Not sure if I answered the question?

Reply:
I think I understand. So a Switch isn't like a conditional renderer. It renders down its component tree to make sure all the routes are rendered?

Reply:
The routing happens at run-time, there is no dictionary of routes to components.

Reply:
Switch happens at run-time

Reply:
Thanks Karthik

Reply:
I think I may have misunderstood the question completely - let me know if that's the case.

Reply:
Oh it's okay. I've worked out what was confusing me.

Reply:
No worries. I just matched a lot of your questions to the [docs](https://reactrouter.com/en/main/start/concepts). Those were good questions.
Re: routing with full paths versus relative-links, react has support for both -  we need to look for paths that don't start with `/` for relative links.. we don't do that here.. we could certainly do that though. Our route configs match the entire path starting with leading `/`.
Re: rendering routes (creating a tree, etc.) to match the paths, that happens during the render of the component. When App.tsx is rendered, the route config for App.tsx alone is processed, converted to a tree and matched.
When PlantingSitesWrapper is rendered, the same thing happens for its specific route config.
That is my understanding at least :)